backend/python/app/middlewares/validate.py

```diff
@@ -0,0 +1,27 @@
+from flask import jsonify, request
+from functools import wraps
+
+
+def validate_request(dto_class_name):
+    """
+    Determine if request is valid based on the types of the arguments passed in to create or update a dto
+
+    :param dto_class_name: the class name to create or update dto
+    :type dto_class_name: str
+    """
+
+    def validate_dto(api_func):
+        @wraps(api_func)
+        def wrapper(*args, **kwargs):
+            dto = eval(dto_class_name)(**request.json)
```

Comment:
Did some research and it turns out `eval` should generally be avoided due to security reasons, let's change this line to:
```python
dto = dtos[dto_class_name](**request.json)
```

where `dtos` is a dictionary defined in the same file
```python
dtos = {
    "CreateUserDTO": CreateUserDTO,
    "EntityDTO": EntityDTO,
    "UpdateUserDTO": UpdateUserDTO
}
```